src/Button/styles.ts

```diff
@@ -22,7 +22,11 @@ export const getVariantStyles = (variant: VariantType = 'default', theme?: Theme
       },
       '&:disabled': {
         color: 'primer.fg.disabled',
-        backgroundColor: 'btn.disabledBg'
+        backgroundColor: 'btn.disabledBg',
+        '[data-component="ButtonCounter"]': {
```

Comment:
If it helps to make the code look nicer, you don't need the quotes:

`[data-component="ButtonCounter"]` is equivalent to `[data-component=ButtonCounter]`

Reply:
Nice! I guess the quotes are only needed if there's special characters in the selector there.